sr_hardware_interface/include/sr_hardware_interface/tactile_sensors.hpp

```diff
@@ -395,11 +401,38 @@ class UBI0PalmData
   boost::array<uint16_t, 16ul> palm;
 };
 
+class MSTData :
+        public GenericTactileData
+{
+public:
+  MSTData() :
+    GenericTactileData()
+  {
+  };
+
+  explicit MSTData(const GenericTactileData &gtd) :
+    GenericTactileData(gtd.tactile_data_valid, gtd.sample_frequency,
+                       gtd.manufacturer, gtd.serial_number,
+                       gtd.software_version_current,
+                       gtd.software_version_server,
+                       gtd.software_version_modified,
+                       gtd.pcb_version)
+  {
+  };
+
+  ~MSTData()
+  {
+  };
+
+  sr_robot_msgs::MSTAll sensor_data;
```

Comment:
This doesn't look right.
The other Data structures don't have the XXXAll msg as part of their data structure. Why would the MST have it?
The XXXData seems to contain data for one tactile sensor unit. Whereas the MSTAll msg contains data for N (normally 5) MST units.